aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/UpdateHandler.java

```diff
@@ -103,7 +105,15 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                         logger
                 ))
                 .then(progress -> updateTags(proxy, rdsProxyClient, progress, previousTags, desiredTags))
-                .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, rdsProxyClient, ec2ProxyClient, logger));
+                .then(progress -> {
+                    desiredResourceState.setTags(Translator.translateTagsFromSdk(Tagging.translateTagsToSdk(desiredTags)));
```

Comment:
I think this might fail detecting drift in resource tags since we always set them from desired model.

Reply:
It compares the desired state with all provided tags (including system and stack tags) and whatever comes from the describe handler. I.e. currently observed Vs desired.